### DISCUSSION TITLE
weed fix without stopping volume server

### DISCUSSION THREAD

**Discussion by bradmurray** [seaweedfs/seaweedfs#6271] [General]
Is there a reason we can't do the following to fix a single volume:
```sh
volid=1234
echo "lock; volume.unmount -node weedvol:9334 -volumeId $volid; unlock" | weed shell -master=weedmaster:9333
weed fix -volumeId=$volid /mnt/weed
echo "lock; volume.mount -node weedvol:9334 -volumeId $volid; unlock" | weed shell -master=weedmaster:9333
```
instead of stopping the volume server for the duration of the process? This is assuming no duplication on the volume.

**Comment by chrislusf** [comment]
Seems fine. But not generic. 
* The `.idx` file location may be on different directory.
* As you mentioned, replication may cause issue.

**Comment by bradmurray** [comment]
I do keep my idx files in a different place, but weed fix doesn't know how to handle that so I just move the file afterwards. Is the existing idx used for any reason? If so there is no way to specify its location.
